core/ArchiveProcessor/Loader.php

```diff
@@ -373,6 +374,44 @@ public function canSkipThisArchive()
         $isArchivingForcedWhenNoVisits = $this->shouldArchiveForSiteEvenWhenNoVisits();
         $hasSiteVisitsBetweenTimeframe = $this->hasSiteVisitsBetweenTimeframe($idSite, $params->getPeriod());
         $hasChildArchivesInPeriod = $this->dataAccessModel->hasChildArchivesInPeriod($idSite, $params->getPeriod());
+        $hasSegment = !$params->getSegment()->isEmpty();
+
+        if ($hasSegment) {
+            $periodEnd = $params->getPeriod()->getDateEnd();
+            $segmentHash = $params->getSegment()->getHash();
+            $segmentArchiving = StaticContainer::get(SegmentArchiving::class);
+            $segmentInfo = $segmentArchiving->findSegmentForHash($segmentHash, $idSite);
+
+            if ($segmentInfo) {
+                $segmentArchiveStartDate = $segmentArchiving->getReArchiveSegmentStartDate($segmentInfo);
+
+                if ($segmentArchiveStartDate->isLater($periodEnd)) {
+                    $doneFlag = Rules::getDoneStringFlagFor(
+                        [$idSite],
+                        $params->getSegment(),
+                        $params->getPeriod()->getLabel(),
+                        $params->getRequestedPlugin()
+                    );
+
+                    $table = Common::prefixTable('archive_invalidations');
+                    $sql = "SELECT MAX(idinvalidation) FROM `$table` WHERE idsite = ? AND date1 = ? AND date2 = ? AND `period` = ? AND `name` = ?";
+
+                    $idInvalidation = Db::fetchOne($sql, [
+                        $idSite,
+                        $params->getPeriod()->getDateStart()->toString(),
```

Comment:
not sure if the timezone is right... might need to be adjusted under circumstances

Reply:
When we run the invalidate report data command we create the ranges in UTC. 
Is there any place where it's possible we should adjust this date? We don't use timestamps in the DB for these fields, only date, even if the timezone different, we can't really adjust it, can we?

Reply:
it's only dates so there shouldn't be a need to apply timezone